pkg/resolution/resolver/git/resolver.go

```diff
@@ -83,6 +87,9 @@ func (r *Resolver) Initialize(ctx context.Context) error {
 	r.logger = logging.FromContext(ctx)
 	r.cache = cache.NewLRUExpireCache(cacheSize)
 	r.ttl = ttl
+	if r.clientFunc == nil {
+		r.clientFunc = factory.NewClient
+	}
```

Comment:
nit: is this conditional needed, or is this function called only from real code?

Reply:
Correct, that's to make sure we actually do the normal client creation outside of tests.

Reply:
sorry I mean can you just do `r.clientFunc = factory.NewClient` (without the if)

Reply:
No - I tried that - I forgot that `Initialize` is called before each `Resolve` call, so it gets overwritten etc.